pkg/validator/sarif.go

```diff
@@ -0,0 +1,49 @@
+package validator
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/xeipuuv/gojsonschema"
+)
+
+type SarifValidator struct{}
+
+func (SarifValidator) Validate(b []byte) (bool, error) {
+	var report map[string]interface{}
+	err := json.Unmarshal(b, &report)
+	if err != nil {
+		customErr := getCustomErr(b, err)
+		return false, customErr
+	}
+
+	schemaUrl, ok := report["$schema"]
+	if !ok {
+		return false, errors.New("error - no schema specified")
+	}
+
+	loadedSchema := gojsonschema.NewReferenceLoader(schemaUrl.(string))
+	loadedReport := gojsonschema.NewRawLoader(report)
+
+	schema, err := gojsonschema.NewSchema(loadedSchema)
+	if err != nil {
+		return false, errors.New("error - schema isn't valid")
```

Comment:
I would wrap the schema in the error message.

Something like


```suggestion
		return false, fmt.Errorf("error - schema isn't valid: %s", schemaURL)
```